packages/hoppscotch-app/helpers/shortcodes/ShortCodeListAdapter.ts

```diff
@@ -0,0 +1,82 @@
+import * as E from "fp-ts/Either"
+import { BehaviorSubject } from "rxjs"
+import { GQLError, runGQLQuery } from "../backend/GQLClient"
+import {
+  GetUserShortCodesQuery,
+  GetUserShortCodesDocument,
+} from "../backend/graphql"
+
+const BACKEND_PAGE_SIZE = 10
+const POLL_DURATION = 10000
+
+export default class ShortCodeListAdapter {
+  error$: BehaviorSubject<GQLError<string> | null>
+  loading$: BehaviorSubject<boolean>
+  shortCodes$: BehaviorSubject<GetUserShortCodesQuery["myShortcodes"]>
+
+  private timeoutHandle: ReturnType<typeof setTimeout> | null
+  private isDispose: boolean
+
+  constructor(deferInit: boolean = false) {
+    this.error$ = new BehaviorSubject<GQLError<string> | null>(null)
+    this.loading$ = new BehaviorSubject<boolean>(false)
+    this.shortCodes$ = new BehaviorSubject<
+      GetUserShortCodesQuery["myShortcodes"]
+    >([])
+    this.timeoutHandle = null
+    this.isDispose = false
+
+    if (!deferInit) this.initialize()
+  }
+
+  initialize() {
+    if (this.timeoutHandle) throw new Error(`Adapter already initialized`)
+    if (this.isDispose) throw new Error(`Adapter has been disposed`)
+
+    const func = async () => {
+      await this.fetchList()
+
+      if (!this.isDispose) {
+        this.timeoutHandle = setTimeout(() => func(), POLL_DURATION)
+      }
+    }
+
+    func()
+  }
+
+  public dispose() {
+    this.isDispose = true
+    clearTimeout(this.timeoutHandle as any)
```

Comment:
TypeScript is raising a valid point here, what if there is no timeout handle (for e.g, already disposed or when not initialized). Should we call `clearTimeout` then ?